packages/odyssey-react/src/components/Icon/Icon.tsx

```diff
@@ -11,49 +11,46 @@
  */
 
 import {
-  Children,
-  cloneElement,
   ComponentPropsWithRef,
   forwardRef,
+  ForwardRefExoticComponent,
 } from "react";
-import type { ReactElement } from "react";
-import { useOid, useOmit } from "../../utils";
+import { useOmit } from "../../utils";
+import type { IconNames } from "./";
+import { iconNameToClassName } from "./";
+import * as Icons from "./";
 
-import styles from "./Icon.module.scss";
+type IconIndexType = keyof typeof Icons;
 
 export interface Props
   extends Omit<ComponentPropsWithRef<"svg">, "style" | "className"> {
   /**
    * Title text used by screen readers
    */
   title?: string;
-
-  children: ReactElement;
+  /**
+   * Name of icon to render
+   */
+  name: IconNames;
 }
 
+type NamedIconProps = Omit<Props, "name">;
+
 /**
  * A system of icons which establishes a visual language
  * that can be easily understood regardless of age, language or culture.
  */
 
 const Icon = forwardRef<SVGSVGElement, Props>(
-  ({ title, children, ...rest }, ref) => {
-    const autoId = "icon_" + useOid();
+  ({ name, title, ...rest }, ref) => {
     const omitProps = useOmit(rest);
 
-    return Children.only(
-      cloneElement(
-        children,
-        {
-          ...omitProps,
-          "aria-labelledby": title && autoId,
-          className: styles.root,
-          ref: ref,
-          role: title ? "img" : "presentation",
-        },
-        [title && <title id={autoId}>{title}</title>, children.props.children]
-      )
-    );
+    const className = iconNameToClassName[name];
+    const NamedIcon = Icons[
+      className as IconIndexType
+    ] as ForwardRefExoticComponent<NamedIconProps>;
```

Comment:
see my note above, but if we're using a type assertion here for each Icon there is a good chance we've lost type information somewhere during the build process where we shouldn't have. Happy to take a look with you soon!

Reply:
for posterity: did a bit of exploration here: https://github.com/okta/odyssey/pull/1090